myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+            for var in currents:
+                self._convert_current_unit(var, helpers=helpers)
+
+            # Convert potentials to mV
+            self._convert_potential_unit(vm)
+
+            time_factor, time_factor_inv = self._get_time_factor(time)
+
+        # Find HH state variables and their alphas, betas, and taus
+        alphas = set()
+        betas = set()
+        taus = set()
+
+        for var in model.states():
+            inf_tau = hh.get_inf_and_tau(var, vm)
+            if inf_tau is not None:
+                taus.add(inf_tau[1])
+                continue
+
+            alpha_beta = hh.get_alpha_and_beta(var, vm)
+            if alpha_beta is not None:
+                alphas.add(alpha_beta[0])
+                betas.add(alpha_beta[1])
+                continue
+
+        # Add time conversion factor for states and HH variables
+        if time_factor is not None:
+            for var in model.states():
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in alphas:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in betas:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in taus:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor))
+
+        # Add intermediary variables on rhs of state derivatives
+        self._prep_derivatives(model)
+
+        return model, currents
+
+    def _generate_diffsl(self, model, extra_out_vars=None):
+        """
+        Generate a DiffSL model from a prepped Myokit model.
+        DiffSL outputs will be set to state variables and extra_out_vars
+        """
+
+        # Create DiffSL-compatible variable names
+        var_to_name = self._create_diffsl_variable_names(model)
+
+        # Create a naming function
+        def var_name(e):
+            if isinstance(e, myokit.LhsExpression):
+                return var_to_name[e.var()]
+            elif isinstance(e, myokit.Variable):
+                return var_to_name[e]
+            raise ValueError(  # pragma: no cover
+                'Not a variable or LhsExpression: ' + str(e)
+            )
+
+        # Create an expression writer
+        e = diffsl.DiffSLExpressionWriter()
+        e.set_lhs_function(var_name)
+
+        export_lines = []  # DiffSL export lines
+        tab = '  '  # Tab character
+
+        # Sort equations in solvable order (grouped by component)
+        sorted_eqs = model.solvable_order()
+
+        # Variables to be excluded from output or handled separately.
+        # Derivatives and their intermediary variables (i.e. dot(x) and dot_x)
+        # are handled in dudt_i, F_i and G_i; time is excluded from the output;
+        # pace is made Vc.
+        time = model.time()
+        pace = model.binding('pace')
+        special_vars = set(
+            [time]
+            + [v.rhs().var() for v in model.states()]
```

Comment:
I don't get this bit. The RHS of a state? Does this mean _all_ state RHS equations have become variables in prep, not just the ones that were referred to by equations?

Reply:
Yes, all state RHS equations are substituted with intermediate variables during prep.

Reply:
Only states with RHS references get intermediary variables in 1a21249